data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -368,16 +368,34 @@ fi
 
 echo "Starting cluster-bootstrap..."
 
-if [ ! -f cb-bootstrap.done ]
-then
-    bootkube_podman_run \
+run_cluster_bootstrap() {
+	bootkube_podman_run \
         --rm \
         --volume "$PWD:/assets:z" \
         --volume /etc/kubernetes:/etc/kubernetes:z \
         "${CLUSTER_BOOTSTRAP_IMAGE}" \
         start --tear-down-early=false --asset-dir=/assets --required-pods="${REQUIRED_PODS}"
+}
 
-    touch cb-bootstrap.done
+gather_bootstrap_on_failure() {
+	source ./bootstrap-cluster-gather.sh
+	cluster_bootstrap_gather
+	if (( $(stat -c%s '/tmp/artifacts-api-server-temp/resources/openapi.json.gz') <= 20 ))
+	then
+		rm -rf /tmp/artifacts-api-server-temp
```

Comment:
This is not the correct behavior. Imagine that the first time through bootkube.sh, the api server artifacts were collected successfully. The second time through bootkube.sh, the control plane did not start, and the api server artifacts were not collected successfully. On the second time through, the code will delete the artifacts gathered, leaving us with nothing. There is also the problem of having the second time through write into the same directory where the artifacts were already collected, polluting the capture with existing files.

1. Gather in a temp location.
2. If the gather was successful, move to a permanent location.
3. In install-gather.sh, after doing a gather and maybe move again, move the artifacts from the permanent location to the gathered bundle.